src/hooks/useQRCodeGenerator.jsx

```diff
@@ -1,78 +1,93 @@
-      // ...
-      // HINT 5: Update the relevant state variables with the generated QR code data URL and adjust the UI accordingly.
-      // ...
-      // HINT 6: Consider the user experience and how the UI should change once the QR code has been generated.
-      // ...
-    );
-  };
+      url,
+      {
+        width: 500,
+        margin: 1,
+        color: {
+          dark: "#335383FF",
+          light: "#EEEEEEFF"
+        }
+      }
+    )
+      .then(url => {
+        setQr(url)
+        setShowInput(false)
+      })
+      .catch(err => console.error(err))
+  }
 
-  // Function to download the generated QR code as a PNG file
+  //Downloads the QRCode as a png with the name the user chooses
   const downloadQRCode = () => {
-    // HINT 1: Consider encapsulating the filename prompting logic into a separate function.
-    const getFileName = () => {
-      // HINT 2: Use a method to prompt the user for input and store the response.
-      // ...
-      // HINT 3: Implement a check for an empty filename and utilize recursion to re-prompt the user if necessary.
-      // ...
-      // HINT 4: Ensure the function returns the obtained filename.
-      // ...
-    };
-
-    // HINT 5: Call the above function to retrieve a filename and store it in a variable.
-    // ...
 
-    // HINT 6: Format the filename to ensure it is filesystem-friendly.
-    // ...
-
-    // HINT 7: Create an anchor element to facilitate the download.
-    // ...
+    const getFileName = () => {
+      let fileName = prompt("Chose a file name for the download:")
 
-    // HINT 8: Set the necessary attributes on the anchor element to prepare it for download.
-    // ...
+      if (fileName == "") 
+        {fileName = prompt("You didn't enter a file name, please do so:")}
 
-    // HINT 9: Append the anchor element to the document to make it interactable.
-    // ...
+      return fileName
+    }
 
-    // HINT 10: Programmatically trigger a click on the anchor element to initiate the download.
-    // ...
+    let theFileName = getFileName()
+    if (theFileName) {
+      theFileName = theFileName.replace(/\s+/g, '-').toLowerCase()
+    
+      //Creates an anchortag which is used to download the image
+      const anchorElement = document.createElement('a')
+      anchorElement.href = qr
+      anchorElement.download = `${theFileName}.png`
+      document.body.appendChild(anchorElement)
 
-    // HINT 11: Remove the anchor element from the document after the download has been initiated.
-    // ...
+      anchorElement.click()
+      document.body.removeChild(anchorElement)
+    }
   };
 
-  // Function to reset the state and allow generating a new QR code
+  // Method to reset the states to allow user to generate a new QR code
   const repeatAction = () => {
-    // Reset the url state to an empty string
-    // Reset the qr state to an empty string
-    // Show the input element back to true :)
+    setUrl(() => "")
+    setQr(() => "")
```

Comment:
I don't think you need to have a method to set the value to "".
Should work with just: 
setUrl("") 
setQR("")